src/main/java/org/fundacionjala/at15/katas/pokerhands/adhemar/Pattern.java

```diff
@@ -0,0 +1,114 @@
+package org.fundacionjala.at15.katas.pokerhands.adhemar;
+
+public abstract class Pattern extends Grouper {
+
+    public Pattern(Card[] cards) {
+        super(cards);
+    }
+
+    public boolean isHighCard() {
```

Comment:
I like your logic. I'm thinking if it would make sense to add one more level to your hierarchy. Say:
```
abstract class Pattern extends Grouper {
     abstract boolean match();
}
```

```
class PairPattern extends Pattern {
    boolean match() {
          // logic to say if group is pair
    }
}
```

Same for the other patterns

Reply:
Great suggestion, I'll work on it

Reply:
Done